packages/sdk/python/human-protocol-sdk/human_protocol_sdk/utils.py

```diff
@@ -219,8 +220,19 @@ def handle_transaction(
             "You must add construct_sign_and_send_raw_middleware middleware to Web3 instance"
         )
     try:
-        tx_hash = tx.transact({"gas": gas_limit or GAS_LIMIT})
+        if tx_options and tx_options.get("gas") is None:
+            tx_options["gas"] = tx.estimate_gas()
+        elif tx_options is None:
+            tx_options = {"gas": tx.estimate_gas()}
+        print(tx_options.get("gas"))
```

Comment:
redundant

Reply:
@alidzm Actually it is not redundant, it is possible that the user does not set any parameter as `tx_options` and in that case it would come as null and would enter in that condition